app/grandchallenge/algorithms/tasks.py

```diff
@@ -163,105 +159,137 @@ def create_algorithm_jobs(
+                # only schedule max_jobs amount of jobs
+                # the rest will be scheduled only after these have succeeded
+                # we do not want to retry the task here, so just stop the loop
+                break
+
+            if len(jobs) >= settings.ALGORITHMS_JOB_BATCH_LIMIT:
+                # raise exception so that we retry the task
+                raise TooManyJobsScheduled
+
+            with transaction.atomic():
+                job = Job.objects.create(
+                    creator=None,  # System jobs, so no creator
+                    algorithm_image=algorithm_image,
+                    algorithm_model=algorithm_model,
+                    algorithm_interface=interface,
+                    task_on_success=task_on_success,
+                    task_on_failure=task_on_failure,
+                    time_limit=time_limit,
+                    requires_gpu_type=requires_gpu_type,
+                    requires_memory_gb=requires_memory_gb,
+                    extra_viewer_groups=extra_viewer_groups,
+                    extra_logs_viewer_groups=extra_logs_viewer_groups,
+                    input_civ_set=ai.values.all(),
+                )
+                on_commit(job.execute)
 
-            jobs.append(job)
+                jobs.append(job)
 
     return jobs
 
 
-def filter_civs_for_algorithm(*, civ_sets, algorithm_image, algorithm_model):
+def filter_archive_items_for_algorithm(
+    *, archive_items, algorithm_image, algorithm_model=None
+):
     """
-    Removes sets of civs that are invalid for new jobs
+    Removes archive items that are invalid for new jobs.
+    The archive items need to contain values for all inputs of one of the algorithm's interfaces.
 
     Parameters
     ----------
-    civ_sets
-        Iterable of sets of ComponentInterfaceValues that are candidate for
-        new Jobs
+    archive_items
+        Archive items whose values are candidates for new jobs' inputs
     algorithm_image
         The algorithm image to use for new job
     algorithm_model
         The algorithm model to use for the new job or None
 
     Returns
     -------
-    Filtered set of ComponentInterfaceValues
+    Dictionary of valid ArchiveItems for new jobs, grouped by AlgorithmInterface
     """
     from grandchallenge.algorithms.models import Job
+    from grandchallenge.evaluation.models import (
+        get_archive_items_for_interfaces,
+        get_valid_jobs_for_interfaces_and_archive_items,
+    )
 
-    input_interfaces = {*algorithm_image.algorithm.inputs.all()}
+    if not archive_items:
+        return {}
 
-    existing_jobs = {
-        frozenset(j.inputs.all())
-        for j in Job.objects.filter(
-            algorithm_image=algorithm_image, algorithm_model=algorithm_model
+    archives = {ai.archive for ai in archive_items}
+    if len(archives) != 1:
+        raise RuntimeError(
+            "The archive items should all belong to the same archive."
```

Comment:
Is there a reason for this restriction? I don't think we ever send that, but if it is due to the logic of the algorithm below then that should be mentioned in a comment (the why). If we filter the jobs by `archive_item` rather than `archive` then I don't think it is.